### PR TITLE
Re coding

### PR DESCRIPTION
**Pull request by Bassem0gamal**

### CODE REVIEW

File: lib/home/home_controller.dart [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
 import 'package:get/get.dart';
-import 'package:getx_test_1/data_source/in_memory_task_data_source.dart';
 
 import '../data_source/task_data_source.dart';
+import '../locator.dart';
 import '../task/models/task_model.dart';
 
 class HomeController extends GetxController {
-  final TaskDataSource _dataSource = InMemoryTaskDataSource.instance;
+  final TaskDataSource _dataSource = locator<TaskDataSource>();
```
Comment (ahmedgemi) [inline lib/home/home_controller.dart]: This is a dependency and should be injected

File: lib/task/add_screen/add_task_controller.dart [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:getx_test_1/data_source/task_data_source.dart';
+import 'package:getx_test_1/task/add_screen/add_task.dart';
+import 'package:intl/intl.dart';
+
+import '../../locator.dart';
+
+class AddTaskController extends GetxController {
+  final TaskDataSource _dataSource = locator<TaskDataSource>();
```
Comment (ahmedgemi) [inline lib/task/add_screen/add_task_controller.dart]: Dependencies should be injected via the constructor

File: lib/task/edit_screen/edit_task_controller.dart [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:getx_test_1/data_source/task_data_source.dart';
+import 'package:getx_test_1/task/edit_screen/edit_task.dart';
+import 'package:intl/intl.dart';
+import '../../locator.dart';
+import '../models/task_model.dart';
+
+class EditTaskController extends GetxController {
+  final TaskDataSource _dataSource = locator<TaskDataSource>();
```
Comment (ahmedgemi) [inline lib/task/edit_screen/edit_task_controller.dart]: Same, Dependencies should be injected

**Review by ahmedgemi — APPROVED** [review]

**Comment by Bassem0gamal** [comment]
Done

**Pull request merged**